app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -0,0 +1,522 @@
+                                createConsolidatedList()
+                            } else {
+                                getImageContributions()
+                            }
+                        }
+                    }
+                    .subscribe({ summary ->
+                        contributionObject.description = StringUtils.defaultString(summary.description)
+                        contributionObject.imageUrl = summary.thumbnailUrl.toString()
+                        contributionObject.pageViews = pageViewsMap[contributionObject.title] ?: 0
+                    }) { t: Throwable? ->
+                        L.e(t)
+                    })
+        }
+    }
+
+    private fun getImageContributions() {
+        disposables.add(ServiceFactory.get(WikiSite(Service.COMMONS_URL)).getUserContributions(AccountUtil.getUserName()!!, 50, imageContributionsContinuation)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ mwQueryResponse ->
+                    continuedImageContributions.clear()
+                    var imageCount = 0
+                    imageContributionsContinuation = if (mwQueryResponse.continuation().isNullOrEmpty()) "" else mwQueryResponse.continuation()!!["uccontinue"]
+                    for (userContribution in mwQueryResponse.query()!!.userContributions()) {
+                        val strArr = userContribution.comment.split(" ")
+                        var contributionLanguage = ""
+                        var editType: Int = -1
+
+                        for (str in strArr) {
+                            if (str.contains("wbsetlabel")) {
+                                val descArr = str.split("|")
+                                if (descArr.size > 1) {
+                                    contributionLanguage = descArr[1]
+                                }
+                                editType = EDIT_TYPE_IMAGE_CAPTION
+                            } else if (str.contains("wbsetclaim")) {
+                                editType = EDIT_TYPE_IMAGE_TAG
+                            }
+                        }
+                        val con = Contribution("", userContribution.title, "", editType, "", DateUtil.iso8601DateParse(userContribution.timestamp),
+                                WikiSite.forLanguageCode(contributionLanguage), 0, if (editType == EDIT_TYPE_IMAGE_TAG) 1 else 0)
+                        if (!continuedImageContributions.add(con)) {
+                            continuedImageContributions.elementAt(continuedImageContributions.indexOf(con)).tagCount++
+                        }
+                    }
+                    for (contribution in continuedImageContributions) {
+                        disposables.add(ServiceFactory.get(WikiSite(Service.COMMONS_URL)).getImageInfo(contribution.title, contribution.wikiSite.languageCode())
```

Comment:
Would it be possible to use `Observable.fromIterable(continuedArticlesContributions).flatMap{...}` for this instead of using for loop?